client/landing/stepper/declarative-flow/internals/steps-repository/design-carousel/index.tsx

```diff
@@ -1,18 +1,100 @@
 import { Design } from '@automattic/design-picker';
-import { StepContainer } from '@automattic/onboarding';
+import { useLocale } from '@automattic/i18n-utils';
+import { ECOMMERCE_FLOW, StepContainer, isLinkInBioFlow } from '@automattic/onboarding';
+import { useMediaQuery } from '@wordpress/compose';
 import { useDispatch } from '@wordpress/data';
 import { useI18n } from '@wordpress/react-i18n';
+import { StarterDesigns, useStarterDesignsQuery } from 'calypso/../packages/data-stores/src';
 import AsyncLoad from 'calypso/components/async-load';
 import FormattedHeader from 'calypso/components/formatted-header';
 import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import type { Step } from '../../types';
 import './style.scss';
 
+const shouldOnlyDisplayMobileCarousel = ( flow: string | null | undefined ) => {
+	switch ( true ) {
+		case isLinkInBioFlow( flow ):
+			return true;
+		default:
+			return false;
+	}
```

Comment:
As a minor question, why use the `switch( true )` here rather than a more direct early return? i.e. something like the following:
```js
	if ( flow && isLinkInBioFlow( flow ) ) {
		return true;
	}

	return false;
}